src/python/tests/annotation/test_update_description.py

```diff
@@ -0,0 +1,307 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.annotation.load` module.
+"""
+
+from contextlib import nullcontext as no_raise
+from pathlib import Path
+from typing import ContextManager
+from sqlalchemy import text
+
+import pytest
+from pytest import CaptureFixture, param, raises
+
+from ensembl.io.genomio.annotation import get_core_data, load_descriptions
+from ensembl.core.models import Gene, Transcript, ObjectXref, Xref, metadata
+from ensembl.utils.database import UnitTestDB
+
+
+def add_gene(dialect: str, session, gene_data=dict) -> None:
+    """Add a gene and a transcript from a gene_data dict.
+
+    Args:
+    dialect: if "mysql", allow to skip foreign key check.
+    session: SQLalchemy session.
+    gene_data:
+        "gene_name" -> gene.stable_id
+        "gene_desc" -> gene.description
+        "tr_name" -> transcript.stable_id
+        "tr_desc" -> transcript.description
+        "gene_xref" -> xref display_name attached to the gene
+    """
+    gene_name = gene_data.get("gene_name", "gene1")
+    gene_description = gene_data.get("gene_desc", "")
+    gene_xref = gene_data.get("gene_xref", "")
+    tr_name = gene_data.get("tr_name", "tr1")
+    tr_description = gene_data.get("tr_desc", "")
+    if dialect == "mysql":
+        session.execute(text("SET FOREIGN_KEY_CHECKS=0"))
+    new_transcript = Transcript(
+        gene_id=1,
+        transcript_id=1,
+        stable_id=tr_name,
+        biotype="protein_coding",
+        description=tr_description,
+        analysis_id=1,
+        seq_region_id=1,
+        seq_region_start=1,
+        seq_region_end=100,
+        seq_region_strand=1,
+        source="test_source",
+    )
+
+    new_gene = Gene(
+        gene_id=1,
+        stable_id=gene_name,
+        biotype="protein_coding",
+        description=gene_description,
+        analysis_id=1,
+        seq_region_id=1,
+        seq_region_start=1,
+        seq_region_end=100,
+        seq_region_strand=1,
+        source="test_source",
+        canonical_transcript_id=1,
+    )
+    session.add(new_gene)
+    session.add(new_transcript)
+
+    if gene_xref:
+        new_xref = Xref(xref_id=1, dbprimary_acc=gene_xref, display_label=gene_xref, external_db_id=1)
+        new_object_xref = ObjectXref(xref_id=1, ensembl_object_type="gene", ensembl_id=1)
+        session.add(new_xref)
+        session.add(new_object_xref)
+
+    session.commit()
+    if dialect == "mysql":
+        session.execute(text("SET FOREIGN_KEY_CHECKS=1"))
+
+
+@pytest.fixture(name="test_db", scope="module")
```

Comment:
```suggestion
@pytest.fixture(name="annot_test_db", scope="module")
```
Just to avoid typos that would be hard to debug, i.e. `test_dbs` is a fixture already provided by `ensembl-utils`.